tests/integration/dynamic_ip_test.py

```diff
@@ -576,6 +606,71 @@ def test_ipv6_autoconf_only(dhcpcli_up):
     libnmstate.apply(desired_state)
 
 
+@pytest.mark.xfail(reason='DHCP server does not run behind vlan', strict=True)
+def test_linux_bridge_over_vlan_over_bond_over_slave_single_transaction(
+    bond0, bond0_vlan101
+):
+    bridge_name = TEST_BRIDGE
+    vlan_ifname = bond0_vlan101[Interface.KEY][0][Interface.NAME]
+    bridge_config_state = add_port_to_bridge(
+        create_bridge_subtree_state(), vlan_ifname, BRIDGE_PORT_CONFIG
+    )
+
+    bridge_ip_config = {
+        Interface.IPV4: create_ipv4_state(enabled=True, dhcp=True)
+    }
+    with linux_bridge(
+        bridge_name,
+        bridge_config_state,
+        extra_iface_state=bridge_ip_config,
+        create=False,
+    ) as bridge0:
+        desired_state = bond0
+        vlan_iface_state = bond0_vlan101[Interface.KEY][0]
+        desired_state[Interface.KEY].append(vlan_iface_state)
+        bridge_iface_state = bridge0[Interface.KEY][0]
+        desired_state[Interface.KEY].append(bridge_iface_state)
+        libnmstate.apply(desired_state)
+
+        desired_state = bridge0
+        desired_state[Interface.KEY][0][Interface.IPV4] = create_ipv4_state(
+            enabled=True, dhcp=True
+        )
+        libnmstate.apply(desired_state)
```

Comment:
bridge0 is already supposed to include the IPv4 state and in general I do not understand why this section is needed.